File: src/utils/logger.ts
import type { LogType, LoggerType } from '@/types/logger';

/* eslint no-console: "off" */
export function logger(type: LoggerType, title: string, message: string | unknown): void {
  const logColors: LogType = {
    SUCCESS: '\x1b[32m',
    INFO: '\x1b[36m',
    WARNING: '\x1b[33m',
    ERROR: '\x1b[31m',
  };

  const logEmoji: LogType = {
    SUCCESS: '✅',
    INFO: '',
    WARNING: '⚠️',
    ERROR: '❌',
  };

  console.log(logColors[type]);
  console.log(`[${title}] ${logEmoji[type]} ${message}`);
  console.log('\x1b[0m');
  if (['ERROR', 'WARNING'].includes(type)) console.log(message);
}

File: pkg/utils/logger.go
package utils

import "fmt"

const (
	Reset  = "\033[0m"
	Red    = "\033[31m"
	Green  = "\033[32m"
	Yellow = "\033[33m"
	Blue   = "\033[36m"
)

type AppLogger struct{}

func (AppLogger) Success(title string, message string) {
	fmt.Printf(Green+"\n[%s] ✅ %s %s\n", title, message, Reset)
}

func (AppLogger) Info(title string, message string) {
	fmt.Printf(Blue+"\n[%s] ℹ️ %s %s\n", title, message, Reset)
}

func (AppLogger) Warn(title string, message string) {
	fmt.Printf(Yellow+"\n[%s] ⚠️ %s %s\n", title, message, Reset)
}

func (AppLogger) Error(title string, message string, err error) {
	fmt.Printf(Red+"\n[%s] ❌ %s %s\n", title, message, err)
}

var Logger = AppLogger{}
